Makefile

```diff
@@ -1,157 +1,74 @@
-EXEC_DIR = executables/
-TAPE_DIR = Images/tapes/
-GIFS_DIR = Images/gifs/
-PACKAGE_NAME = tcping
-VERSION = 2.6.0
-ARCHITECTURE = amd64
-DEB_PACKAGE_DIR = $(EXEC_DIR)debian
-DEBIAN_DIR = $(DEB_PACKAGE_DIR)/DEBIAN
-CONTROL_FILE = $(DEBIAN_DIR)/control
-EXECUTABLE_PATH = $(EXEC_DIR)/tcping
-TARGET_EXECUTABLE_PATH = $(DEB_PACKAGE_DIR)/usr/bin/
-PACKAGE = $(PACKAGE_NAME)_$(ARCHITECTURE).deb
-MAINTAINER = https://github.com/pouriyajamshidi
-DESCRIPTION = Ping TCP ports using tcping. Inspired by Linux's ping utility. Written in Go
-
-.PHONY: all build update clean format test vet gitHubActions container \
-		create_dirs build_linux_amd64 build_debian_package build_linux_static build_linux_arm \
-		build_windows build_macos build_macos_arm  build_freebsd
+# ==================================================
+# Constants
+# ==================================================
+
+# Meta
+SHELL := /bin/bash
+VERSION := 2.6.0
+MAINTAINER := https://github.com/pouriyajamshidi
+DESCRIPTION := Ping TCP ports using tcping. Inspired by Linux's ping utility. Written in Go
+
+# IO directories
+TARGET_DIR := target
+OUTPUT_DIR := output
+TAPES_DIR := Images/tapes
+GIFS_DIR := Images/gifs
+
+# File lists
+RELEASE_ARTIFACTS := \
+	$(OUTPUT_DIR)/tcping-freebsd-amd64-static.tar.gz \
+	$(OUTPUT_DIR)/tcping-freebsd-amd64-dynamic.tar.gz \
+	$(OUTPUT_DIR)/tcping-freebsd-arm64-static.tar.gz \
+	$(OUTPUT_DIR)/tcping-freebsd-arm64-dynamic.tar.gz \
+	$(OUTPUT_DIR)/tcping-linux-amd64-static.tar.gz \
+	$(OUTPUT_DIR)/tcping-linux-amd64-dynamic.tar.gz \
+	$(OUTPUT_DIR)/tcping-linux-arm64-static.tar.gz \
+	$(OUTPUT_DIR)/tcping-linux-arm64-dynamic.tar.gz \
+	$(OUTPUT_DIR)/tcping-darwin-amd64-static.tar.gz \
+	$(OUTPUT_DIR)/tcping-darwin-amd64-dynamic.tar.gz \
```

Comment:
`darwin-amd64` is already included, unless by `darwin-x86` he's referring to the 32 bit `i686` target (why?).

Reply:
Indeed, he is referring to the 32-bit.
Not exactly sure why, but I think they want to support very old devices.

Reply:
> Indeed, he is referring to the 32-bit. Not exactly sure why, but I think they want to support very old devices.

I'll go double check with him. He's asking for something that existed for about 6 months, almost 20 years ago [[ref]](https://en.wikipedia.org/wiki/Mac_transition_to_Intel_processors#32-bit_regression).

Supporting such old targets is very risky from a maintainability perspective - it's very foreseeable (and unfortunately, common) that the toolchain or some library may suddenly stop working one day, and nobody wants to fix it because nobody cares. He better have a ***very good*** reason.

Reply:
I think more likely, he's referring to x64. Currently the MacOS x64 binary is dynamically linked; I think he wants a static one.

Reply:
Thank you for following it up with him :)